css/src/atomics/width.scss

```diff
@@ -0,0 +1,101 @@
+$widths: 100, 150, 200, 250, 300, 350, 400, 450, 500 !default;
+// do we need a maximum for globally applicable widths?
+// i.e. .width-500 on a 414px wide screen would be a problem
+// do we need min and max? They are used sometimes.
+// But we're definitely generating more of every category of atomic than we need.
+
+// widths
+
+@each $width in $widths {
+	.width-#{$width} {
+		width: #{$width}px !important;
+	}
+}
+
+@include tablet {
+	@each $width in $widths {
+		.width-#{$width}-tablet {
+			width: #{$width}px !important;
+		}
+	}
+}
+
+@include desktop {
+	@each $width in $widths {
+		.width-#{$width}-desktop {
+			width: #{$width}px !important;
+		}
+	}
+}
+
+@include widescreen {
+	@each $width in $widths {
+		.width-#{$width}-widescreen {
+			width: #{$width}px !important;
+		}
+	}
+}
+
+// max-widths
+
+@each $width in $widths {
+	.max-width-#{$width} {
+		max-width: #{$width}px !important;
+	}
+}
+
+@include tablet {
+	@each $width in $widths {
+		.max-width-#{$width}-tablet {
+			max-width: #{$width}px !important;
+		}
+	}
+}
+
+@include desktop {
+	@each $width in $widths {
+		.max-width-#{$width}-desktop {
+			max-width: #{$width}px !important;
+		}
+	}
+}
+
+@include widescreen {
+	@each $width in $widths {
+		.max-width-#{$width}-widescreen {
+			max-width: #{$width}px !important;
+		}
+	}
+}
+
+// min-widths
+
+@each $width in $widths {
+	.min-width-#{$width} {
+		min-width: #{$width}px !important;
+	}
+}
+
+@include tablet {
+	@each $width in $widths {
+		.min-width-#{$width}-tablet {
+			min-width: #{$width}px !important;
+		}
+	}
+}
+
+@include desktop {
+	@each $width in $widths {
+		.min-width-#{$width}-desktop {
+			min-width: #{$width}px !important;
+		}
+	}
+}
+
+@include widescreen {
+	@each $width in $widths {
+		.min-width-#{$width}-widescreen {
+			min-width: #{$width}px !important;
+		}
+	}
+}
```

Comment:
Is the output size of the css reduced by restructuring the code as follows? Also thinking this could be better in terms of ordering for responsive use?

Having looked through the docs-ui code using `has(-min|-max)?-width-.*`, many of the min-width usages could use width and many of the max-width usages have to do with truncation and could use flex grow/shrink to control width of the element being truncated.

```scss
@each $width in $widths {
  // width loop
  // min-width loop
  // max-width loop    
}

@include tablet {
  // width loop
  // min-width loop
  // max-width loop    
}

@include desktop {
  // width loop
  // min-width loop
  // max-width loop    
}

@include widescreen {
  // width loop
  // min-width loop
  // max-width loop    
}
```

Reply:
Team discussed in a meeting. Think we landed on the following:

-  no min width rules, they tend not to be necessary
- need a good strategy for unsetting global widths on tablet +. The width equivelent to something like `display-none display-none-tablet`. This strategry could be based on `unset`, `auto`, or `inherit` but need to do some research to figure out what's best.
- we'll need to be careful that globally applicable widths do no exceed common mobile boundaries, or that wider ones are scoped to tablet. This could be a confusing an unexpected thing. First consideration is to preclude mobile with bugs where `.width-400` for example exceed an iPhone SE's (or whatever small-ish or old phone's) horizontal boundaries.